lib/terminal.coffee

```diff
@@ -41,7 +41,8 @@ module.exports = class Terminal extends EventEmitter
     "#{protocol}://#{@host}:#{@port}/#{@path}?token=#{@token}"
 
   reset: ->
-    @socket.connect()
+    @socket.close()
+    setTimeout(@connect.bind(this), 100)
```

Comment:
not totally sure about this—without the timeout, a reset just wouldn't happen, but with it, it will actually reconnect